Updated OpenAir Spec

lsalehar opened the pull request:
Updated the OpenAIR file format specification:

- Added the required header:
  - Version format
  - Author
- Added the unclassified airspace class
- Extended airspace types
- Added airspace activation times command: AA.
- Added command to support airspace transponder codes: AX. Closes #8

lsalehar commented:
Hey @reskume please have a look.

akolarck commented:
@llauner please have a look at this Pull request. Comments are welcome.

akolarck commented:
@lsalehar There's a typo in the ISO number. Correct number is 8601. The link is correct, just the text has this typo

akolarck commented:
@reskume If this was a new format I would certainly agree to make type required. But to keep backwards compatibility I believe we can't. I don't see how we can make it required in the sense that we would fail loading an airspace if AY type definition is missing.

Everyone who makes new OpenAir files is of course well advised to use AY Type definition in addition to class.

reskume commented:
Yeah, backwards compatibility should be provided whenever possible but in this particular case, unfortunately, I would argue to make a hard cut and make this a V2 (or whatever name fits best) version of openAIR. Making the **AY** optional is 100% possible but it would be against the actual definition of the format as it is defined currently, because **AC** only allows classes as per this proposed definition... Or maybe I misunderstand something. But, honestly, every current openAIR puts everything into the **AC** tags and it is pure chaos. To have a better format, I suppose their is a "one time investment" for each maintainer of a file to port it to this new format definition. Otherwise, this could get ugly very soon because this new version will again be a very relaxed/loosely defined format.

When finalized, I can update our [OpenAIR Format Fixer](https://github.com/openAIP/openaip-openair-fix-format) to convert standard OpenAIR v1 into the new "v2" format. And even if the tool is not used, adding the AY tag is not that much work. I had to do it manually for some of our source data files and it is acceptable, at least from my point of view.

llauner commented on OpenAir_File_Format_Support.md:
```diff
@@ -29,15 +29,30 @@ Geographic coordinates can be represented in two formats:
 ### Distance
 
-Distance is measured in nautical miles `nm` and is implied, meaning it is specified <u>without a unit.</u>  This applies to definitions such as radii, widths and segments, see: [DA](#da:-define-an-arc-between-start-and-end-bearing), [DB](#db:-define-an-arc-between-start-and-end-point), [DY](#dy:-define-an-airway-segment), or [V](#v:-variable-assignment) for more.
+Distance is measured in nautical miles `nm` andis implied, meaning it is specified <u>without a unit.</u>  This applies to definitions such as radii, widths and segments, see: [DA](#da:-define-an-arc-between-start-and-end-bearing), [DB](#db:-define-an-arc-between-start-and-end-point), [DY](#dy:-define-an-airway-segment), or [V](#v:-variable-assignment) for more.
```
Missing space between andis -> and is

lsalehar replied:
Fixed in the lat commit.

llauner commented on OpenAir_File_Format_Support.md:
```diff
@@ -95,38 +113,65 @@ AN ED-R6 Brokdorf H24
 
 #### AY: Airspace Type
```
I agree with that. AC + AY should be required

llauner commented on OpenAir_File_Format_Support.md:
```diff
@@ -95,38 +113,65 @@ AN ED-R6 Brokdorf H24
+| MTR     | Military Training Route                                    |
+| OFR     | Overflight Restriction                                     |
+| P       | Prohibited Area                                            |
+| PROT    | Protected Airspace                                         |
```
What's a Protected airspace ? Is that for bird protection area for instance ?

llauner reviewed (changes requested):
Added a few comments...thanks for the update

akolarck commented:
@reskume @llauner We decided to make AY+AC mandatory for v2 specification of the OpenAir protocol. @lsalehar will modify the text accordingly.

Feel free to prepare your parsers accordingly.

reskume commented:
This is absolutely great news!

reskume commented:
Just as an idea: if we define a new v2 format now we can also think about having two other commands that would provide max and min ceilings. Currently we can provide this information with OpenAIP JSON format but not with the OpenAIR format as it is now (and also not with the v2). Is this something we should consider? Any thoughts on that?

reskume commented on OpenAir_File_Format_Support.md:
```diff
@@ -160,6 +207,27 @@ AG Nordholz Radar
+
+#### AA: Airspace Activation Times
+
+*Optional.* Use [ISO8601](https://en.wikipedia.org/wiki/ISO_8601) time interval format to express the time when the airspace is active. Only the time interval format is allowed and must be specified in UTC (Zulu) time, no local or time offsets are supproted. `NONE` token can be used to indicate the unspecified start or end time of the airspace activation. In case of `None/None`  this airspace is  ...
```
Typo: supproted

lsalehar replied:
Fixed in the latest commit.

reskume commented on OpenAir_File_Format_Support.md:
```diff
@@ -160,6 +207,27 @@ AG Nordholz Radar
+
+#### AA: Airspace Activation Times
+
+*Optional.* Use [ISO8601](https://en.wikipedia.org/wiki/ISO_8601) time interval format to express the time when the airspace is active. Only the time interval format is allowed and must be specified in UTC (Zulu) time, no local or time offsets are supproted. `NONE` token can be used to indicate the unspecified start or end time of the airspace activation. In case of `None/None`  this airspace is  ...
```
Use `NONE/NONE` instead of allowing both uppercase and lowercase

lsalehar replied:
Fixed in the latest update

reskume commented:
> @reskume @llauner We decided to make AY+AC mandatory for v2 specification of the OpenAir protocol. @lsalehar will modify the text accordingly.
> 
> Feel free to prepare your parsers accordingly.

I updated our parser to support the new proposed version 2 specification: [https://github.com/openAIP/openaip-openair-parser](https://github.com/openAIP/openaip-openair-parser)

xtophe80 commented on OpenAir_File_Format_Support.md:
```diff
@@ -160,6 +207,27 @@ AG Nordholz Radar
+
+#### AT: Airspace Activation Times
+
+*Optional.* Use [ISO8061](https://en.wikipedia.org/wiki/ISO_8601) time interval format to express the time when the airspace is active. Only the time interval format is allowed and must be specified in UTC (Zulu) time, no local or time offsets are supproted. `NONE` token can be used to indicate the unspecified start or end time of the airspace activation. In case of `None/None`  this airspace is  ...
```
Would it be good to include a way to specify week days like Monday-Friday or Saturday-Sunday. Handling of public holidays (Saturday+Sunday+Public holidays) would be great but another layer of complexity. Days could be combined with time like FridayT20:00Z/MondayT08:00Z

lsalehar replied:
We purposefully decided against this, exactly because of the problem you mention. We haven't found many examples that have a nicely structured operating hours and days, also the airspaces can be updated more easily nowadays, due to the propagation of internet connectivity in the cockpit.

lsalehar commented:
For all concerned, please have a look at the document again. I have made the following changes:
- Fixed the typos
- Made AY required
- Updated AY Tokens table:
   - Removed GP, PROTECTED, TRP 
   - Added NONE, TFR (As US Temporary Flight Restriction), ATZ, CUSTOM, N, TRZ, VFRR (VFR Route) 
   - Renamed  TFR -> TRAFR

reskume reviewed (commented):
Besides the still pending adjustment to use AMSL instead of MSL for lower alt limit I have nothing to add at the moment.

lsalehar commented on OpenAir_File_Format_Support.md:
```diff
@@ -160,6 +207,27 @@ AG Nordholz Radar
+
+#### AA: Airspace Activation Times
+
+*Optional.* Use [ISO8601](https://en.wikipedia.org/wiki/ISO_8601) time interval format to express the time when the airspace is active. Only the time interval format is allowed and must be specified in UTC (Zulu) time, no local or time offsets are supproted. `NONE` token can be used to indicate the unspecified start or end time of the airspace activation. In case of `None/None`  this airspace is  ...
```
Fixed in the latest commit.

lsalehar commented on OpenAir_File_Format_Support.md:
```diff
@@ -160,6 +207,27 @@ AG Nordholz Radar
+
+#### AA: Airspace Activation Times
+
+*Optional.* Use [ISO8601](https://en.wikipedia.org/wiki/ISO_8601) time interval format to express the time when the airspace is active. Only the time interval format is allowed and must be specified in UTC (Zulu) time, no local or time offsets are supproted. `NONE` token can be used to indicate the unspecified start or end time of the airspace activation. In case of `None/None`  this airspace is  ...
```
This was introduced really for convenience of people maintaining the OpenAIR files. I would argue that AA NONE/NONE is exclusive and other AA are ignored if they appear together. I added this clarification into the latest revision.

lsalehar commented on OpenAir_File_Format_Support.md:
````diff
@@ -191,7 +259,7 @@ Format:
 ```
 
-Altitude references must be `AGL`, `FL`, `STD`, `MSL`. For areas extending to the ground, use `GND`, omitting the altitude.
+Altitude references are `AGL`, `FL`, `STD`, `MSL`. For areas extending to the ground, use `GND`, omitting the altitude.
````
Agreed. Fixed
